iocore/net/quic/QUICGlobals.cc

```diff
@@ -85,6 +85,23 @@ QUIC::ssl_client_new_session(SSL *ssl, SSL_SESSION *session)
   return 0;
 }
 
+void
+QUIC::ssl_client_keylog_cb(const SSL *ssl, const char *line)
+{
+  QUICTLS *qtls           = static_cast<QUICTLS *>(SSL_get_ex_data(ssl, QUIC::ssl_quic_tls_index));
+  const char *keylog_file = qtls->keylog_file();
+  std::ofstream file(keylog_file, std::ios_base::app);
```

Comment:
How many times will this function be called? Doesn't it overwrite this file every time?

Reply:
9 times for each session. The `app` open mode seeks to the eos before write, iiuc.

Reply:
So it open and close the file 9 times, it doesn't sound great.
Also, in that case, does it mean I need to remove the file before I start new session?

Reply:
I agree it's no ideal, but it not big deal for traffic_quic. For origin server side connection or server side connection, this could be problem. But when we support it, it needs some changes to dump keys for each connections. And `proxy.config.quic.client.session_file` has same problem. So these should be fixed in same time and it's out of scope from this PR.

As long as you're using wireshark, you don't need to remove. It can find the correct key.

Reply:
Then please create an issue or put a TODO comment for the future work at least. I'm not going to stop you but I don't like leaving a trap like this.

Reply:
Roger.

Reply:
Filed as #5263